cmd/updater/update.sh

```diff
@@ -138,8 +138,88 @@ function determine_current_version() {
     echo Current Version = ${CURRENTVER}
 }
 
+function get_updater_url() {
+    local UNAME
+    local OS
+    local ARCH
+    UNAME=$(uname)
+    if [[ "${UNAME}" = "Darwin" ]]; then
+        OS="darwin"
+        UNAME=$(uname -m)
+        if [[ "${UNAME}" = "x86_64" ]]; then
+            ARCH="amd64"
+        else
+            echo "This platform ${UNAME} is not supported by updater."
+            exit 1
+        fi
+    elif [[ "${UNAME}" = "Linux" ]]; then
+        OS="linux"
+        UNAME=$(uname -m)
+        if [[ "${UNAME}" = "x86_64" ]]; then
+            ARCH="amd64"
+        elif [[ "${UNAME}" = "armv6l" ]]; then
+            ARCH="arm"
+        elif [[ "${UNAME}" = "armv7l" ]]; then
+            ARCH="arm"
+        elif [[ "${UNAME}" = "aarch64" ]]; then
+            ARCH="arm64"
+        else
+            echo "This platform ${UNAME} is not supported by updater."
+            exit 1
+        fi
+    else
+        echo "This operation system ${UNAME} is not supported by updater."
+        exit 1
+    fi
+    UPDATER_FILENAME="install_master_${OS}-${ARCH}.tar.gz"
+    UPDATER_URL="https://github.com/algorand/go-algorand-doc/raw/master/downloads/installers/${OS}_${ARCH}/${UPDATER_FILENAME}"
+}
+
+# check to see if the binary updater exists. if not, it will automatically the correct updater binary for the current platform
+function check_for_updater() {
+    # check if the updater binary exist.
+    if [ -f "${SCRIPTPATH}/updater" ]; then
+        return 0
+    fi
+    get_updater_url
+
+    # check the curl is available.
+    CURL_VER=$(curl -V 2>/dev/null || true)
+    if [ "${CURL_VER}" = "" ]; then
+        # no curl is installed.
+        echo "updater binary is missing and cannot be downloaded since curl is missing."
+        if [[ "$(uname)" = "Linux" ]]; then
+            echo "To install curl, run the following command:"
+            echo "apt-get update; apt-get install -y curl"
+        fi
+        exit 1
+    fi
+
+    CURL_OUT=$(curl -LJO --silent ${UPDATER_URL})
```

Comment:
I confused. 
1. -J tells curl to use Content-Disposition for output file name instead of URL name. But after we refer the file as UPDATER_FILENAME from URL. Can't Content-Disposition and actual file name be different?
2. curl drops file into current directory but we looking the file at "${SCRIPTPATH}/${UPDATER_FILENAME}" I do not see where SCRIPTPATH is set as CWD.